### PR TITLE
Progress tracking

### PR DESCRIPTION
**Pull request by jayoshih**

### CODE REVIEW

File: ricecooker/__main__.py [hunk 1]
```diff
@@ -1,33 +1,49 @@
...
         kwarg = arg.split('=')
         kwargs.update({kwarg[0].strip(): kwarg[1].strip()})
       except IndexError:
         raise InvalidUsageException("Invalid kwarg '{0}' found: Must format as [key]=[value] (no whitespace)".format(arg))
-    if arguments["--debug"]:
-    	domain = config.DEBUG_DOMAIN
-    uploadchannel(arguments["<file_path>"], domain, verbose=arguments["-v"], **kwargs)
\ No newline at end of file
+
+    step = arguments['--step']
```
Comment (jayoshih) [inline ricecooker/__main__.py]: Get step to start at

File: setup.py [hunk 2]
```diff
@@ -1,18 +1,25 @@
...
-install_reqs = parse_requirements('requirements.txt')
 
-with open('README.rst') as readme_file:
+with open('README.md') as readme_file:
     readme = readme_file.read()
 
 with open('docs/history.rst') as history_file:
     history = history_file.read()
 
-requirements = [str(ir.req) for ir in install_reqs]
```
Comment (jayoshih) [inline setup.py]: problems with reading requirements.txt

File: ricecooker/config.py [hunk 3]
```diff
@@ -18,45 +22,89 @@
...
+
+
+def authentication_url(domain):
+    """ authentication_url: returns url to login to Kolibri Studio
+        Args: domain (str): domain to log in
+        Returns: string url to authenticate_user_internal endpoint
+    """
+    return AUTHENTICATION_URL.format(domain=domain)
+
+def init_file_mapping_store(debug):
```
Comment (jayoshih) [inline ricecooker/config.py]: Code for storing/loading from a pickle to get last state of ricecooker

File: ricecooker/config.py [hunk 4]
```diff
@@ -18,45 +22,89 @@
 CREATE_CHANNEL_URL = "{domain}/api/internal/create_channel"
 
 # URL to return after channel is created
-OPEN_CHANNEL_URL = "{domain}/open_channel/{invitation_id}/{channel_id}"
+OPEN_CHANNEL_URL = "{domain}/channels/{channel_id}/edit"
```
Comment (jayoshih) [inline ricecooker/config.py]: No longer rely on invitations to accept invitations

File: ricecooker/commands.py [hunk 5]
```diff
@@ -1,64 +1,174 @@
...
-def uploadchannel(path, domain, verbose=False, **kwargs):
     """ uploadchannel: Upload channel to Kolibri Studio server
         Args:
             path (str): path to file containing channel data
-            domain (str): domain to upload to
+            debug (bool): determine which domain to upload to
             verbose (bool): indicates whether to print process (optional)
         Returns: (str) link to access newly created channel
     """
+    # Get domain to upload to
```
Comment (jayoshih) [inline ricecooker/commands.py]: Get correct domain

File: ricecooker/commands.py [hunk 6]
```diff
@@ -1,64 +1,174 @@
...
+    domain = config.PRODUCTION_DOMAIN
+    if debug:
+      domain = config.DEBUG_DOMAIN
+    config.init_file_mapping_store(debug)
+
+    if verbose:
+        print("\n\n***** Starting channel build process *****")
+
+    # Set up progress tracker
+    progress_manager = RestoreManager(debug)
```
Comment (jayoshih) [inline ricecooker/commands.py]: Get last session if requested by user

File: ricecooker/classes/questions.py [hunk 7]
```diff
@@ -159,6 +169,87 @@ def validate(self):
         except AssertionError as ae:
             raise InvalidQuestionException("Invalid question: {0}".format(self.__dict__))
 
+    def process_question(self, downloader):
```
Comment (jayoshih) [inline ricecooker/classes/questions.py]: Go through question, hints, and answers and get images

File: ricecooker/__main__.py [hunk 8]
```diff
@@ -1,33 +1,49 @@
-"""Usage: ricecooker uploadchannel [-hvqr] <file_path> [--debug] [[OPTIONS] ...]
```
Comment (jayoshih) [inline ricecooker/__main__.py]: Update docopt

File: README.md [hunk 9]
```diff
@@ -100,6 +100,9 @@ A sample program has been created [here](https://github.com/learningequality/ric
 	Once you have created the appropriate question model, add it to an exercise model with `<exercise-node>.add_question(<question>)`
 
 * Running the Rice Cooker
-	Run `python -m ricecooker uploadchannel [-v] "<path-to-py-file>" [--debug]`
```
Comment (jayoshih) [inline README.md]: Update documentation

File: ricecooker/classes/questions.py [hunk 10]
```diff
@@ -80,11 +82,8 @@ def process_question(self, downloader):
             hint_files += hfiles
         self.hints = hints
 
-        # Process raw data
-        self.raw_data, data_files = self.set_images(self.raw_data, downloader)
-
         # Return all files
-        return question_files + answer_files + hint_files + data_files
+        return question_files + answer_files + hint_files
```
Comment (jayoshih) [inline ricecooker/classes/questions.py]: Consolidate all images processed

File: ricecooker/sample_program.py [hunk 11]
```diff
@@ -4,98 +4,78 @@
 from le_utils.constants import content_kinds,file_formats, format_presets, licenses, exercises
 
 SAMPLE_PERSEUS = '{"answerArea":{"chi2Table":false,"periodicTable":false,"tTable":false,"zTable":false,"calculator":false},' + \
-'"hints":[{"widgets":{},"images":{},"content":"Hint #1","replace":false},{"widgets":{},"images":{},"content":"Hint #2","replace":false}],' +\
+'"hints":[{"widgets":{},"images":{"web+graphie:storage/0a0c0f1a1a40226d8d227a07dd143f8c08a4b8a5": {}},"content":"Hint #1","replace":false},{"widgets":{},"images":{},"content":"Hint #2","replace":false}],' +\
```
Comment (jayoshih) [inline ricecooker/sample_program.py]: Changes to test exercises more

File: ricecooker/commands.py [hunk 12]
```diff
@@ -1,64 +1,174 @@
...
+    else:
+        channel_link = progress_manager.channel_link
+
+    # Open link on web browser (if specified) and return new link
+    print("DONE: Channel created at {0}".format(channel_link))
+    prompt_open(channel_link)
+    progress_manager.set_done()
+    return channel_link
+
+def prompt_token(domain):
```
Comment (jayoshih) [inline ricecooker/commands.py]: Prompt user to enter authentication token

**Comment by aronasorman** [comment]
I'll wait for this to get merged, so the common commits won't get displayed on the other PRs :)

**Pull request merged**